File: java/src/test/IceCreamTestDrive.java
package test;

import main.java.mx.iteso.Cone;
import main.java.mx.iteso.stores.MilkStore;
import main.java.mx.iteso.stores.WaterStore;
import main.java.mx.iteso.stores.YoghurtStore;

public class IceCreamTestDrive {

	public static void main(String[] args) {
		MilkStore milkStore = new MilkStore();
		WaterStore waterStore = new WaterStore();
		YoghurtStore yoghurtStore = new YoghurtStore();
		
		Cone cone = milkStore.prepareCone("Mango");
		
		System.out.println("First cone is: " + cone.iceCream.name);
		System.out.println();
		System.out.println();
		
		cone = waterStore.prepareCone("Strawberry");
		
		System.out.println("Second cone is: " + cone.iceCream.name);
		System.out.println();
		System.out.println();
		
		cone = yoghurtStore.prepareCone("Guava");
		System.out.println("Third cone is: " + cone.iceCream.name);
		System.out.println();
		System.out.println();
		
		cone = waterStore.prepareCone("Pineapple");
		System.out.println("Fourth cone is: " + cone.iceCream.name);
		System.out.println();
		System.out.println();	
		

	}

}

File: csharp/factory_method/test/IceCreamTestDrive.cs
using factory_method.main;
using factory_method.main.stores;
using System;
using System.Collections.Generic;
using System.Text;

namespace factory_method.test
{
    class IceCreamTestDrive
    {
        public static void Main(String[] args)
        {
            MilkStore milkStore = new MilkStore();
            WaterStore waterStore = new WaterStore();
            YoghurtStore yoghurtStore = new YoghurtStore();

            Cone cone = milkStore.prepareCone("Mango");

            Console.WriteLine("First cone is: " + cone.getIceCream().getName());
            Console.WriteLine();
            Console.WriteLine();

            cone = waterStore.prepareCone("Strawberry");

            Console.WriteLine("Second cone is: " + cone.getIceCream().getName());
            Console.WriteLine();
            Console.WriteLine();

            cone = yoghurtStore.prepareCone("Guava");
            Console.WriteLine("Third cone is: " + cone.getIceCream().getName());
            Console.WriteLine();
            Console.WriteLine();

            cone = waterStore.prepareCone("Pineapple");
            Console.WriteLine("Fourth cone is: " + cone.getIceCream().getName());
            Console.WriteLine();
            Console.WriteLine();
        }
        
    }
}
